packages/suite/src/components/suite/modals/confirm/ConfirmValueOnDevice.tsx

```diff
@@ -50,25 +51,44 @@ const StyledDeviceDisconnected = styled(DeviceDisconnected)`
 `;
 
 export interface ConfirmDeviceScreenProps extends Pick<ModalProps, 'onCancel' | 'heading'> {
-    device: TrezorDevice;
     copyButtonText: React.ReactNode;
     copyButtonDataTest?: string;
     isConfirmed?: boolean;
     value: string;
+    verify: () => ThunkAction;
     valueDataTest?: string;
 }
 
 export const ConfirmValueOnDevice = ({
     copyButtonText,
     copyButtonDataTest,
-    device,
     heading,
     isConfirmed,
+    verify,
     onCancel,
     value,
     valueDataTest,
 }: ConfirmDeviceScreenProps) => {
     const dispatch = useDispatch();
+    const { device } = useDevice();
+
+    useEffect(() => {
+        const verifyAndContinue = async () => {
+            if (!device?.available) {
+                const result = await dispatch(applySettings({ use_passphrase: true }));
+                if (!result || !result.success) return;
+            }
+            if (onCancel) onCancel();
+            dispatch(verify());
+        };
+
+        if (device?.connected) verifyAndContinue();
```

Comment:
not sure here but what happens if you connect the device and it will not be acquired? meaning it will be in use by another application at that moment.

Reply:
Hmm I don't even know that this state is with Trezor possible. What other check do you suggest so we can overcome this state?